### ISSUE TITLE
Gradient checkpointing

### ISSUE THREAD

**Issue by nicolas-dufour** [chuanyangjin/fast-DiT#5]
Hi, can you point out where does the gradient checkpointing is happening in the code? I cannot figure it out.

Thanks!

**Comment by chuanyangjin** [comment]
In models.py, on line 251, we employ a ckpt wrapper to encapsulate the forward pass of each DiTBlock module. This
guarantees that all subsequent calls employ checkpointing. Then we use 'torch.utils.checkpoint' to incorporate gradient checkpointing into the process.

**Comment by nicolas-dufour** [comment]
I see thanks!

**Comment by thecodingwizard** [comment]
Just curious, is the checkpoint wrapper necessary? If I just use

```
for block in self.blocks:
    x = torch.utils.checkpoint.checkpoint(block, x, c) 
```

it seems to behave the same.

Thanks for releasing this repository!
